.github/workflows/test-rules-engine-functional.yml

```diff
@@ -0,0 +1,30 @@
+name: Test Rules Engine (Functional)
+
+on: [push]
+
+defaults:
+  run:
+    working-directory: rules-engine
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10"]
+
+    steps:
+      - uses: actions/checkout@v3

```

Comment:
`suggestion`: we could split the first set of steps from these workflows ...

```lang=yaml
    steps:
      - uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          python -m pip install --upgrade pip
          pip install -r requirements-dev.txt
          pip install -e .
```

into a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) to avoid duplication.